frontend/pages/hosts/ManageHostsPage/ManageHostsPage.jsx

```diff
@@ -147,24 +149,48 @@ export class ManageHostsPage extends PureComponent {
       isAllMatchingHostsSelected: false,
       searchQuery: "",
       hosts: [],
-      isHostsLoading: false,
-      isTeamsLoading: true,
-      sortBy: [],
+      isHostsLoading: true,
+      sortBy: [{ id: DEFAULT_SORT_HEADER, direction: DEFAULT_SORT_DIRECTION }],
+      isConfigLoaded: null,
+      isTeamsLoaded: null,
+      isBasicTier: null,
     };
   }
 
   componentDidMount() {
-    const { dispatch, isBasicTier } = this.props;
+    const { dispatch } = this.props;
     dispatch(getLabels());
-    if (isBasicTier) {
-      dispatch(teamActions.loadAll({}));
-    }
   }
 
-  componentDidUpdate(prevProps) {
-    const { dispatch, isBasicTier } = this.props;
-    if (isBasicTier !== prevProps.isBasicTier && isBasicTier) {
-      dispatch(teamActions.loadAll({}));
+  componentWillReceiveProps() {
+    const { config, dispatch } = this.props;
+    const { isConfigLoaded } = this.state;
+
+    if (!isConfigLoaded && !isEmpty(config)) {
+      const isBasicTier = permissionUtils.isBasicTier(config);
+      if (isBasicTier) {
+        dispatch(teamActions.loadAll({}))
+          .then(() => {
+            this.setState({
+              isBasicTier,
+              isTeamsLoaded: true,
+            });
+          })
+          .catch((error) => {
+            renderFlash(
+              "error",
+              "An error occured loading teams. Please try again."
+            );
+            console.log(error);
+            this.setState({
+              isBasicTier,
+              isTeamsLoaded: false,
+            });
+          });
+      } else {
+        this.setState({ isBasicTier });
```

Comment:
Generally it's good to avoid copying props into state, as this confuses the "source of truth" for the data.

Reply:
I definitely try to observe that rule in general and was thinking maybe this particular case fell within the exception to the rule that Dave Ceddia outlines. In this case, isBasicTier is derived from the config prop so I couldn't find a way to reliably have an initial undefined or null state (here the config prop is typically empty on page load so the initial derived state is false) as a means to effectively gate premature API calls and rendering of other stateful subcomponents that was producing undesirable UX in various places like the host table.

Reply:
Can you throw a 15min meeting on my calendar to have a look at the specific UX issues and see if we can come up with a way to work around them without adding to state? Thanks!